### PR TITLE
Allow kwargs passed to open() with overrides

### PR DESCRIPTION
**Pull request by martindurant**

### CODE REVIEW

File: fsspec/core.py [hunk 1]
```diff
@@ -456,6 +456,8 @@ def open(
     - For implementations in separate packages see
       https://filesystem-spec.readthedocs.io/en/latest/api.html#other-known-implementations
     """
+    kw = {"expand": False}
+    kw.update(kwargs)
```
Comment (Skylion007) [inline fsspec/core.py:460]: i would use dict.setdefault in the future. :)

File: fsspec/implementations/local.py [hunk 2]
```diff
@@ -127,6 +127,14 @@ def cp_file(self, path1, path2, **kwargs):
         else:
             raise FileNotFoundError(path1)
 
+    def isfile(self, path):
```
Comment (Skylion007) [inline fsspec/implementations/local.py:130]: We really never supported these? 0o0

Comment (martindurant) [reply fsspec/implementations/local.py:130]: Previously this was supported via a call to info() in the superclass, which does stat to get sizes and permissions, and so is slower.

File: fsspec/core.py (hunk 1 above)
Comment (martindurant) [reply fsspec/core.py:460]: Following the conversation, `expand` should probably be promoted to the signature of open() to help those wanting to be explicit.

**Pull request merged**